modules/consensus/synchronize.go

```diff
@@ -45,121 +42,121 @@ func (g *Gateway) threadedResynchronize() {
 // such transmissions may be required to fully synchronize.
 //
 // TODO: don't run two Synchronize threads at the same time
-func (g *Gateway) Synchronize(peer modules.NetAddress) error {
-	g.log.Println("INFO: synchronizing to", peer)
-	for {
+func (s *State) Synchronize(peer modules.NetAddress) error {
+	// loop until there are no more blocks available
+	moreAvailable := true
+
+	for moreAvailable {
```

Comment:
not my preferred style

2 statement for loops are legal? I didn't even know that